weather_mv/loader_pipeline/sinks.py

```diff
@@ -175,18 +175,22 @@ def _get_band_data(i):
             raise RuntimeError("Wrong regex passed in --forecast_time_regex.")
         ds.attrs['start_time'] = start_time
         ds.attrs['end_time'] = end_time
+    elif 'start_time' in ds.attrs and 'end_time' in ds.attrs:
```

Comment:
3 points
- Our tiff files do not preserve attrs so reading them with `xr.open_dataset(file.tiff, engine='rasterio')` will have empty `ds.attrs`.
- 'start_time' and 'end_time' in dataset attrs is in `2021-01-01T00:00:00Z` format so we need to parse it into datetime.datetime() for below datetime_value_s variable to initialize.
- `start_time` is not being used. Can be skipped.

Reply:
Good points Deep.
May be we can pick this https://github.com/google/weather-tools/issues/159#issuecomment-1342490273 next & restructure `_preprocess_tif` altogether.